File: solutions/python/secret-handshake/2/secret_handshake.py
def commands(binary_str):
    actions = ['wink', 'double blink', 'close your eyes', 'jump']
    
    handshake = [action for index, action in enumerate(actions) if int(binary_str, 2) & 2 ** index]
    
    return handshake[::-1] if int(binary_str, 2) & 2 ** 4 else handshake
        

File: solutions/javascript/secret-handshake/2/secret-handshake.js
export const commands = number => {
  const possibleActions = [
    'wink',
    'double blink',
    'close your eyes',
    'jump'
  ]
  
  const actions = [];
  
  for (let i = 0; i < 4; i++) {
    if (number & 2 ** i) actions.push(possibleActions[i])
  }
  
  return number & 16 ? actions.reverse() : actions;
};
